src/ledger/LedgerTxn.cpp

```diff
@@ -263,6 +287,35 @@ LedgerTxn::Impl::create(LedgerTxn& self, LedgerEntry const& entry)
     return ltxe;
 }
 
+void
+LedgerTxn::createOrUpdateWithoutLoading(LedgerEntry const& entry)
+{
+    return getImpl()->createOrUpdateWithoutLoading(*this, entry);
+}
+
+void
+LedgerTxn::Impl::createOrUpdateWithoutLoading(LedgerTxn& self,
+                                              LedgerEntry const& entry)
+{
+    throwIfSealed();
+    throwIfChild();
+
+    auto key = LedgerEntryKey(entry);
+
+    auto current = std::make_shared<LedgerEntry>(entry);
+    auto impl = LedgerTxnEntry::makeSharedImpl(self, *current);
+
+    // Set the key to active before constructing the LedgerTxnEntry, as this
+    // can throw and the LedgerTxnEntry destructor requires that mActive
+    // contains key. LedgerTxnEntry constructor does not throw so this is
+    // still exception safe.
+    mActive.emplace(key, toEntryImplBase(impl));
```

Comment:
Ah, it's only been a few weeks since I wrote the comment explaining what "active" means and I've already forgotten it / started to think it means something else. Ok.

Reply:
discussed in-chat, resolved.